app/src/main/java/org/wikipedia/settings/DeveloperSettingsPreferenceLoader.kt

```diff
@@ -0,0 +1,207 @@
+package org.wikipedia.settings
+
+import android.content.DialogInterface
+import androidx.appcompat.app.AlertDialog
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.schedulers.Schedulers
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.analytics.eventplatform.Event
+import org.wikipedia.analytics.eventplatform.EventPlatformClient
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.dataclient.page.PageSummary
+import org.wikipedia.history.HistoryEntry
+import org.wikipedia.notifications.NotificationPollBroadcastReceiver
+import org.wikipedia.page.PageActivity
+import org.wikipedia.page.PageTitle
+import org.wikipedia.readinglist.database.ReadingListDbHelper
+import org.wikipedia.readinglist.database.ReadingListPage
+import org.wikipedia.setupLeakCanary
+import org.wikipedia.suggestededits.provider.EditingSuggestionsProvider.getNextArticleWithMissingDescription
+import org.wikipedia.talk.TalkPageSeenDatabaseTable.resetAllUnseen
+import org.wikipedia.util.StringUtil
+import org.wikipedia.util.StringUtil.fromHtml
+import java.util.*
+
+internal class DeveloperSettingsPreferenceLoader(fragment: PreferenceFragmentCompat) : BasePreferenceLoader(fragment) {
+    private val setMediaWikiBaseUriChangeListener = Preference.OnPreferenceChangeListener { _, _ ->
+        resetMediaWikiSettings()
+        true
+    }
+    private val setMediaWikiMultiLangSupportChangeListener = Preference.OnPreferenceChangeListener { _, _ ->
+        resetMediaWikiSettings()
+        true
+    }
+
+    override fun loadPreferences() {
+        loadPreferences(R.xml.developer_preferences)
+        setUpMediaWikiSettings()
+        findPreference(R.string.preferences_developer_crash_key).onPreferenceClickListener = Preference.OnPreferenceClickListener { throw TestException("User tested crash functionality.") }
+        findPreference(R.string.preference_key_add_articles).onPreferenceChangeListener = Preference.OnPreferenceChangeListener { _: Preference, newValue: Any ->
+            if (!isEmptyOrZero(newValue)) {
+                createTestReadingList(TEXT_OF_TEST_READING_LIST, 1, StringUtil.trim(newValue).toInt())
```

Comment:
I'm not sure I see why we need our own `trim()` function. (Why can't we use `.toString().trim()`?)

Reply:
Make sense. Done.